packages/connect-react/src/index.ts

```diff
@@ -1,41 +1,46 @@
 import React, {
   createElement,
+  useCallback,
   useContext,
   useEffect,
   useMemo,
   useRef,
   useState,
 } from 'react'
 import {
-  Application,
+  App,
   ConnectOptions,
   ConnectorDeclaration,
   Organization,
   Permission,
   connect,
 } from '@aragon/connect'
 
-type ContextValue = {
-  org: Organization | null
-  orgError: Error | null
-  orgLoading: boolean
-}
-
 type ConnectProps = {
   children: React.ReactNode
   connector: ConnectorDeclaration
   location: string
   options?: ConnectOptions
 }
-
-type OrganizationHookResult = [Organization | null, Error | null, boolean]
-type AppsHookResult = [Application[], Error | null, boolean]
-type PermissionsHookResult = [Permission[], Error | null, boolean]
+type ContextValue = {
+  org: Organization | null
+  orgLoadingStatus: LoadingStatus
+}
+type LoadingStatus = {
+  error: Error | null
+  loading: boolean
+  retry: () => void
```

Comment:
I think this is fine for this version, but since we are dealing with subscriptions, we should probably auto-retry for the next one.

- `error` would still get populated (and emptied on success).
- It would still be possible to call `retry`, e.g. we could imagine a user wanting to retry immediately by clicking a button.
- We could have a `retryEvery` prop on the provider, that either accept a number (milliseconds), or a function implementing a delay based on the number of attempts. Setting it to `-1` would disable the automatic retry behavior.

Reply:
> Since we are dealing with subscriptions

Outside of the `retryEvery` prop or a user calling `retry`, the subscription would only help here if the underlying data was updated in the mean time, right? Or could you see a case where you hadn't connected / had some sort of corruption, and the GraphQL subscription just magically fixed it at a later point?